### ISSUE TITLE
Add Transforms to compute popular remote sensing indices

### ISSUE THREAD

**Issue by isaaccorley** [microsoft/torchgeo#112]
Indices are commonly computed for various applications in remote sensing as linear combinations of existing bands in imagery e.g. [Normalized Difference Vegetation Index (NDVI)](https://www.indexdatabase.de/db/i-single.php?id=58) = `(NIR - RED) / (NIR + RED)`. We should look to add some `nn.Modules` under `torchgeo.transforms` for some popular ones.

- [Remote Sensing Index Database](https://www.indexdatabase.de/db/i.php)
